File: crates/ga-parser/tests/fixtures/kotlin-tiny/android-mini/UserRepository.kt
package com.example

import com.example.User

class UserRepository {
    private val cache: MutableMap<Int, User> = mutableMapOf()

    fun findById(id: Int): User? = cache[id]

    fun save(u: User) {
        cache[u.name.hashCode()] = u
    }

    suspend fun fetchRemote(id: Int): User {
        delay(10)
        return User("remote-$id")
    }
}

suspend fun delay(millis: Long) {
    // marker: suspend fn at top level
}

File: crates/ga-parser/tests/fixtures/csharp-tiny/aspnet-mini/UserRepository.cs
using System.Collections.Generic;
using System.Threading.Tasks;
using App.Models;

namespace App.Data;

public class UserRepository
{
    private readonly Dictionary<int, User> _cache = new Dictionary<int, User>();

    public User FindById(int id)
    {
        return _cache.TryGetValue(id, out var u) ? u : null;
    }

    public void Save(User u)
    {
        _cache[u.Name.GetHashCode()] = u;
    }

    public async Task<User> FetchRemoteAsync(int id)
    {
        await Task.Delay(10);
        return new User($"remote-{id}");
    }
}
